loopy/match.py

```diff
@@ -243,8 +243,19 @@ def __call__(self, kernel, matchable):
 
 class Tagged(GlobMatchExpressionBase):
     def __call__(self, kernel, matchable):
+        from loopy.kernel.instruction import LegacyStringInstructionTag
         if matchable.tags:
-            return any(self.re.match(tag) for tag in matchable.tags)
+            return any(
+                    self.re.match(tag.value)
+                    if isinstance(tag, LegacyStringInstructionTag)
+                    else
+
+                    self.re.match(tag) if isinstance(tag, str)
+                    else
+
+                    False
```

Comment:
So, right now in order to support tag-matching the tags have to be LegacyStringInstruction tags right?

Reply:
That's correct.